.github/workflows/pull-request.yml

```diff
@@ -266,15 +262,35 @@ jobs:
       - uses: ./.github/actions/setup-zsh
       - uses: ./.github/actions/short-sha
 
-      - name: Configure AWS credentials for test account
-        uses: ./.github/actions/configure-aws-credentials
-        with:
-          account-name: 'test'
-          aws-region: ${{ env.AWS_REGION }}
-          test-account-role: ${{ secrets.UHD_TERRAFORM_ROLE_TEST }}
+      - name: Debug AWS Role Selection
+        run: |
+          echo "TARGET_ENV is set to: $TARGET_ENV"
+          echo "Selected AWS_ROLE: $AWS_ROLE"
+
+      - name: Set AWS Role for Target Environment
+        env:
+          TARGET_ENV: ${{ needs.build_base.outputs.target_env || 'test' }}
+          AWS_ROLE_AUTH_DEV: ${{ secrets.UHD_TERRAFORM_ROLE_AUTH_DEV }}
+          AWS_ROLE_AUTH_TEST: ${{ secrets.UHD_TERRAFORM_ROLE_AUTH_TEST }}
+          AWS_ROLE_TEST: ${{ secrets.UHD_TERRAFORM_ROLE_TEST }}
+        run: |
+          if [[ "$TARGET_ENV" == "auth-dev" ]]; then
+            echo "AWS_ROLE=$AWS_ROLE_AUTH_DEV" >> $GITHUB_ENV
```

Comment:
As above, we should only be interacting with either of the test accounts from the PR workflow